### PR TITLE
ensure subject is used instead of userid. 

### PR DESCRIPTION
**Pull request by kpaul**
get identity provider from claims when finding permissions for a user
when searching for user use the userid (subjectid:identityprovider)
added identityprovider to endpoint for getting a users groups

### CODE REVIEW

File: Fabric.Authorization.API/Modules/UsersMetadataModule.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ public class UsersMetadataModule : BaseMetadataModule
         private readonly Parameter _userIdParameter = new Parameter
         {
             Name = "userId",
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersMetadataModule.cs]: Do we also need to change the name of the parameter from userId to subjectId?

File: Fabric.Authorization.API/Modules/UsersModule.cs [hunk 2]
```diff
@@ -37,31 +38,31 @@ public class UsersModule : FabricModule<User>
             Get("/permissions", async _ => await GetCurrentUserPermissions().ConfigureAwait(false), null,
                 "GetUserPermissions");
 
-            Post("/{userId}/AdditionalPermissions",
+            Post("/{subjectId}/AdditionalPermissions",
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersModule.cs]: Do we also need identityProvider here? I think these endpoints could be used by actors who are not the subject/target of the permission changes.

File: Fabric.Authorization.API/Modules/UsersModule.cs [hunk 3]
```diff
@@ -37,31 +38,31 @@ public class UsersModule : FabricModule<User>
             Get("/permissions", async _ => await GetCurrentUserPermissions().ConfigureAwait(false), null,
                 "GetUserPermissions");
 
-            Post("/{userId}/AdditionalPermissions",
+            Post("/{subjectId}/AdditionalPermissions",
                 async param => await this.AddGranularPermissions(param, denied: false).ConfigureAwait(false), null,
                 "AddPermissions");
 
-            Post("/{userId}/DeniedPermissions",
+            Post("/{subjectId}/DeniedPermissions",
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersModule.cs]: I think this endpoint has the same issue as above.

File: Fabric.Authorization.API/Modules/UsersModule.cs [hunk 4]
```diff
@@ -37,31 +38,31 @@ public class UsersModule : FabricModule<User>
...
             return groups;
         }
 
         private async Task<dynamic> AddGranularPermissions(dynamic param, bool denied)
         {
             var apiModel = this.Bind<GranularPermissionApiModel>();
 
-            if (apiModel.Target != param["userId"])
-                return CreateFailureResponse("Target must be the user id.", HttpStatusCode.BadRequest);
+            if (apiModel.Target != param["subjectId"])
```
Comment (michaelvidal24) [inline Fabric.Authorization.API/Modules/UsersModule.cs]: Can we make this a case insensitive comparison? Also - not related to your changes, I am wondering why we make this check, i.e. what is the purpose of the Target property on the GranularPermissionApiModel? @deeiinu can you refresh my memory on the purpose of the Target property?

File: Fabric.Authorization.API/Modules/UsersModule.cs (hunk 2 above)
Comment (kpaul) [reply Fabric.Authorization.API/Modules/UsersModule.cs]: You are getting ahead of me here. That is what I am going to suggest in the api changes to support identity provider that I have not submitted to the team yet.

File: Fabric.Authorization.API/Modules/UsersMetadataModule.cs (hunk 1 above)
Comment (kpaul) [reply Fabric.Authorization.API/Modules/UsersMetadataModule.cs]: That is changed in another pull request where I have the swagger doc updates.

File: Fabric.Authorization.API/Modules/UsersModule.cs (hunk 4 above)
Comment (kpaul) [reply Fabric.Authorization.API/Modules/UsersModule.cs]: I was curious about that myself. I realize that it represents the subjectId of the user to change permissions for but I am not sure why its called Target.

**Pull request merged**